TWS: Add tws and untws tools

Nostritius opened the pull request:
This PR adds wo TheWitcherSave related tools, for packing and unpacking TheWitcherSave files.

DrMcCoy commented:
```
src/archives/util.cpp: In function ‘Common::UString Archives::findPath(const Common::UString&, Aurora::FileType, uint64, Common::HashAlgo)’:
src/archives/util.cpp:54:5: error: suggest explicit braces to avoid ambiguous ‘else’ [-Werror=dangling-else]
  if (!name.empty())
     ^
```

```
src/tws.cpp: In function ‘int main(int, char**)’:
src/tws.cpp:47:18: error: unused variable ‘game’ [-Werror=unused-variable]
   Aurora::GameID game = Aurora::kGameIDUnknown;
                  ^~~~
```

DrMcCoy commented on src/archives/util.cpp:
```diff
@@ -52,7 +52,11 @@ static Common::UString findPath(const Common::UString &name, Aurora::FileType ty
+		if (TypeMan.getFileType(name) == Aurora::kFileTypeNone)
+			path = TypeMan.addFileType(name, type);
+		else
+			path = name;
```
This breaks the unherf tool, because Sonic HERF files contain files with multiple extensions. Namely files compressed with the "small"-algorithm.

Like, for example, itm_ace_timmy_0.ncgr.small is now shown/extracted as itm_ace_timmy_0.ncgr, which is wrong.

Nostritius replied:
Are .small files the only occasion where this happens? Becauseif so i would simply add a condition like
```cpp
if (TypeMan.getFileType(name) == Aurora::kFileTypeNone || type == Aurora::kFileTypeSMALL)
```

DrMcCoy replied:
Yeah, I don't want such a hacky solution :P

DrMcCoy replied:
No, you should not

DrMcCoy replied:
Wait, *why* is that necessary to begin with?

File archives shouldn't carry an extension in their filenames.

I.e. the real, actual issue is that the `TheWitcherSaveFile` Archive class should strip extensions from the filenames, like we're doing with ZIP files, certain ERFs, NDS, HERFs, OBB, ... (see https://github.com/xoreos/xoreos/blob/master/src/aurora/zipfile.cpp#L61, https://github.com/xoreos/xoreos/blob/master/src/aurora/erffile.cpp#L600, https://github.com/xoreos/xoreos/blob/master/src/aurora/herffile.cpp#L133, https://github.com/xoreos/xoreos/blob/master/src/aurora/ndsrom.cpp#L92, https://github.com/xoreos/xoreos/blob/master/src/aurora/obbfile.cpp#L97)

DrMcCoy replied:
And yes, I probably should have caught that earlier, sorry

Nostritius commented:
I have now rewritten the file ending block in a (hopefully) more elegant way. I also fixed the other issues.

Nostritius commented:
I have now switched the writing and reading to not directly use file types in the file names. I also fixed some other issues and modified the unit tests accordingly.

DrMcCoy commented:
Merged as ed047ed...367563e, thanks! :)
